### PR TITLE
datasource for network acl and network acl collection

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMIsNetworkACLDataSourceBasic'
=== RUN   TestAccIBMIsNetworkACLDataSourceBasic
--- PASS: TestAccIBMIsNetworkACLDataSourceBasic (93.58s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 95.790s

$ make testacc TESTARGS='-run=TestAccIBMIsNetworkAclsDataSourceBasic'
=== RUN   TestAccIBMIsNetworkAclsDataSourceBasic
--- PASS: TestAccIBMIsNetworkAclsDataSourceBasic (29.89s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm 32.679s
...
```
Terraform show -> https://ibm.box.com/s/a6pb24owm4ellha1dj1m5k3p0w5uihug

### CODE REVIEW

File: ibm/data_source_ibm_is_network_acl.go [hunk 1]
```diff
@@ -0,0 +1,703 @@
...
+		for _, networkAcl := range allrecs {
+			if *networkAcl.VPC.Name == vpc_name_str && network_acl_name == *networkAcl.Name {
+				networkACL = &networkAcl
+				vpc_found = true
+				break
+			}
+		}
+
+		if !vpc_found {
+			log.Printf("[DEBUG] VPC with name %s not found", vpc_name_str)
```
Comment (hkantare) [inline ibm/data_source_ibm_is_network_acl.go:389]: should the error message no network acl found with given vpc name and acl name?

File: ibm/data_source_ibm_is_network_acl.go [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+			return diag.FromErr(fmt.Errorf("VPC with name %s not found", vpc_name_str))
+		}
+	} else {
+
+		getSubnetNetworkACLOptions := &vpcv1.GetSubnetNetworkACLOptions{}
+
+		getSubnetNetworkACLOptions.SetID(d.Get("network_acl").(string))
+
+		networkACLInst, response, err := vpcClient.GetSubnetNetworkACLWithContext(context, getSubnetNetworkACLOptions)
+		if err != nil {
```
Comment (hkantare) [inline ibm/data_source_ibm_is_network_acl.go:399]: Add if err != nil || networkACLInst== nil

File: website/docs/d/is_network_acl.html.markdown [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  name = ibm_is_network_acl.isExampleACL.name
+  vpc_name = ibm_is_vpc.testacc_vpc.name
+}
+```
+
+## Argument Reference
+
+Review the argument reference that you can specify for your data source.
+
+- `network_acl` - (Optional, String) The network acl identifier.
```
Comment (hkantare) [inline website/docs/d/is_network_acl.html.markdown:39]: Arrange arguments/attributes in alphabetical order

File: ibm/data_source_ibm_is_network_acl.go [hunk 4]
```diff
@@ -0,0 +1,703 @@
...
+
+func dataSourceIBMIsNetworkACL() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMIsNetworkACLRead,
+
+		Schema: map[string]*schema.Schema{
+			isNetworkACLName: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				RequiredWith: []string{"vpc_name"},
```
Comment (kavya498) [inline ibm/data_source_ibm_is_network_acl.go:26]: this should also have this below check
```
ExactlyOneOf: []string{"name", "network_acl"},
```

**Pull request merged**